abel/tools/circularize.py

```diff
@@ -0,0 +1,291 @@
+    if ref_angle == None:
+        factor = np.mean(radial_correction_function(theta))
+    else:
+        factor = radial_correction_function(ref_angle)
+
+    # radial correction
+    Xactual = X*factor/radial_correction_function(theta)
+    Yactual = Y*factor/radial_correction_function(theta)
+
+    # @DanHickstein magic
+    # https://github.com/PyAbel/PyAbel/issues/186#issuecomment-275471271
+    IMcirc = ndimage.interpolation.map_coordinates(IM,
+                                  (origin[1] - Yactual, Xactual + origin[0]))
+
+    return IMcirc
+
+
+def _residual(param, radial, profile, previous):
+    """ `scipy.optimize.leastsq` residuals function.
+
+        Evaluate the difference between a radial-scaled intensity profile
+        and its adjacent angular slice.
+
+    """
+
+    radial_scaling, amplitude = param[0], param[1]
+
+    newradial = radial*radial_scaling
+    spline_prof = UnivariateSpline(newradial, profile, s=0, ext=3)
+    newprof = spline_prof(radial)*amplitude
+
+    # residual cf adjacent slice profile
+    return newprof - previous
+
+
+def correction(polarIMTrans, angles, radial, method):
+    """Evaluate a radial correctionn factor that aligns an angular slice
+       radial intensity profile with its adjacent (previous) slice profile.
+
+
+    Parameters
+    ----------
+    polarIMTrans : numpy 2D array
+        polar coordinate image, transposed (theta, radius) so that each row 
+        is a single angle
+
+    radial : numpy 1D array
+        radial coordinates of one column of polarIMTrans
+
+    angles : numpy 1D array
+        angle coordinates of one row of polarIMTrans
+
+    method : str
+        'lsq' : determine a radial correction factor that will align a
+        radial intensity profile with the previous, adjacent slice.
+        'argmax': radial correction factor from position of maximum intensity
+
+    """
+
+    if method == "argmax":
+        # follow position of intensity maximum
+        pkpos = []
+
+        for ang, aslice in zip(angles, polarIMTrans):
+            profile = aslice
+            pkpos.append(profile.argmax())  # store index of peak position
+
+        # radial correction factor relative to peak max in first angular slice
+        radcorr = radial[pkpos[0]]/radial[pkpos]
+
+    elif method == "lsq":
+        # least-squares radially scale intensity profile to match previous slice
+
+        fitpar = np.array([1.0, 1.0])  # radial scale factor, amplitude
+        radcorr = []
+        radcorr.append(1)  # first slice nothing to compare with
+        previous = polarIMTrans[0]
+
+        for ang, aslice in zip(angles[1:], polarIMTrans[1:]):
+            profile = aslice
+
+            result = leastsq(_residual, fitpar, args=(radial, profile,
+                             previous))
+
+            radcorr.append(result[0][0])  # radial scale factor direct from lsq
+
+            previous += _residual(result[0], radial, profile, previous)
```

Comment:
add comment:
```
# This "previous" slice corresponds to the previous slice that has been re-scaled. Thus, if the next slice is identical, it will be assigned a scale factor of 1.0
```